### ISSUE TITLE
Deeply nested fields are incorrectly serialized

### ISSUE THREAD

**Issue by gerbal** [chargebee/chargebee-typescript#27]
I have a Coupon with nested attributes, which this library does not correctly encoded: 

Given a coupon like: 

```json
{"item_constraints": [{"item_price_ids": ["id1"]}]}
```

Should be serialized as 

```
item_constraints[item_price_ids][0]=["id1"] 
```

However, this library serializes this payload incorrectly

```typescript
Core.encodeParams({item_constraints: [{price_item_ids: ['id1']}]})
// "item_constraints[item_price_ids][0][0]=id1"
```

**Comment by cb-prajaktachavan** [comment]
Closing this issue as its been resolved in latest release v2.6.0.

**Comment by cb-prajaktachavan** [comment]
Hi, You should be passing this jsonarray as string '["id1","id2"]' within single quotes as this field is not list of values.
